ui/src/main/java/org/pentaho/di/ui/spoon/Spoon.java

```diff
@@ -3002,49 +3043,167 @@ public void delHop() {
   }
 
   public void delClusterSchema() {
-    final TransMeta transMeta = (TransMeta) selectionObjectParent;
-    final ClusterSchema clusterSchema = (ClusterSchema) selectionObject;
-    delegates.clusters.delClusterSchema( transMeta, clusterSchema );
+    withClusterSchema( ( manager, clusterSchema ) -> delegates.clusters.delClusterSchema( manager, clusterSchema ) );
   }
 
   public void monitorClusterSchema() throws KettleException {
-    final ClusterSchema clusterSchema = (ClusterSchema) selectionObject;
-    monitorClusterSchema( clusterSchema );
+    withClusterSchema( ( clusterSchemaManager, clusterSchema ) -> monitorClusterSchema( clusterSchema ) );
+  }
+
+  public void moveClusterSchemaToGlobal() {
+    withClusterSchema( ( clusterSchemaManager, clusterSchema ) -> {
+      getLog().logBasic( "Moving the cluster schema " + clusterSchema.getName() + " to global");
+      delegates.clusters.moveToGlobal( clusterSchemaManager, clusterSchema );
+    } );
+  }
+
+  public void moveClusterSchemaToProject() {
+    withClusterSchema( ( clusterSchemaManager, clusterSchema ) -> {
+      getLog().logBasic( "Moving the cluster schema " + clusterSchema.getName() + " to project");
+      delegates.clusters.moveToProject( clusterSchemaManager, clusterSchema );
+    } );
+  }
+
+  public void copyClusterSchemaToGlobal() {
+    withClusterSchema( ( clusterSchemaManager, clusterSchema ) -> {
+      getLog().logBasic( "Copying the cluster schema " + clusterSchema.getName() + " to global");
+      delegates.clusters.copyToGlobal( clusterSchemaManager, clusterSchema );
+    } );
   }
 
+  public void copyClusterSchemaToProject() {
+    withClusterSchema( ( clusterSchemaManager, clusterSchema ) -> {
+      getLog().logBasic( "Copying the cluster schema " + clusterSchema.getName() + " to project");
+      delegates.clusters.copyToProject( clusterSchemaManager, clusterSchema );
+    } );
+  }
+
+  public void dupeClusterSchema() {
+    withClusterSchema( ( clusterSchemaManager, clusterSchema ) -> {
+      getLog().logBasic( "Duplicating the cluster schema " + clusterSchema.getName() );
+      delegates.clusters.dupeClusterSchema( clusterSchemaManager, clusterSchema );
+    } );
+  }
+
+
+  /**
+   * Edit the Slave server configuration
+   */
   public void editSlaveServer() {
-    final SlaveServer slaveServer = (SlaveServer) selectionObject;
-    editSlaveServer( slaveServer );
+    SpoonTreeLeveledSelection leveledSelection = (SpoonTreeLeveledSelection) selectionObject;
+    editSlaveServer( leveledSelection );
+  }
+
+  private void editSlaveServer( SpoonTreeLeveledSelection leveledSelection ) {
+    withSlaveServer( leveledSelection, ( slaveServerManager, slaveServer ) ->
+    {
+      getLog().logBasic( "editing slave server " + slaveServer.getName() );
+      delegates.slaves.edit( slaveServerManager, slaveServer );
+    } );
+  }
+
+  private <T> T getSharedObjectManager( LeveledTreeNode.LEVEL level, Class<T> clazz ) throws KettleException {
```

Comment:
Can you add a javadoc comment for Spoon#getSharedObjectManager. I believe this is an important entry into project profile logic.